Mathlib/NumberTheory/Padics/RingHoms.lean

```diff
@@ -178,14 +178,24 @@ theorem exists_mem_range : ∃ n : ℕ, n < p ∧ x - n ∈ maximalIdeal ℤ_[p]
   apply max_lt hr
   simpa using hn
 
+theorem exists_unique_mem_range : ∃! n : ℕ, n < p ∧ x - n ∈ maximalIdeal ℤ_[p] := by
+  obtain ⟨n, hn₁, hn₂⟩ := exists_mem_range x
+  use n, ⟨hn₁, hn₂⟩, fun m ⟨hm₁, hm₂⟩ ↦ ?_
+  have := (zmod_congr_of_sub_mem_max_ideal x n m hn₂ hm₂).symm
+  rwa [ZMod.natCast_eq_natCast_iff, ModEq, mod_eq_of_lt hn₁, mod_eq_of_lt hm₁] at this
+
 /-- `zmod_repr x` is the unique natural number smaller than `p`
 satisfying `‖(x - zmod_repr x : ℤ_[p])‖ < 1`.
 -/
 def zmodRepr : ℕ :=
-  Classical.choose (exists_mem_range x)
+  Classical.choose (exists_unique_mem_range x).exists
 
 theorem zmodRepr_spec : zmodRepr x < p ∧ x - zmodRepr x ∈ maximalIdeal ℤ_[p] :=
-  Classical.choose_spec (exists_mem_range x)
+  Classical.choose_spec (exists_unique_mem_range x).exists
+
+theorem zmodRepr_unique (y : ℕ) (hy : y < p ∧ x - y ∈ maximalIdeal ℤ_[p]) : y = zmodRepr x :=
```

Comment:
Could you please split `hy` into two assumptions?

Reply:
Done, thanks!